src/Microsoft.DotNet.Cli.Utils/ProcessReaper.cs

```diff
@@ -0,0 +1,163 @@
+using System;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.PlatformAbstractions;
+using Microsoft.Win32.SafeHandles;
+
+using RuntimeEnvironment = Microsoft.DotNet.PlatformAbstractions.RuntimeEnvironment;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    /// <summary>
+    /// Responsible for reaping a target process if the current process terminates.
+    /// </summary>
+    /// <remarks>
+    /// On Windows, a job object will be used to ensure the termination of the target
+    /// process (and its tree) even if the current process is rudely terminated.
+    ///
+    /// On POSIX systems, the reaper will handle SIGTERM and attempt to forward the
+    /// signal to the target process only.
+    ///
+    /// The reaper also suppresses SIGINT in the current process to allow the target
+    /// process to handle the signal.
+    /// </remarks>
+    internal class ProcessReaper : IDisposable
```

Comment:
Guys, y’ know…that…that…it doesn't work for me. I gotta have more cowbell!

Reply:
Don't fear the (process) reaper, Nick.

Reply:
It's going to be in my head all week now. :)

Reply:
Worse, the image is going to stay in my head.